src/AddOns/BassFx/Effects/Objects/VolumeEffect.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ManagedBass.Fx
+{
+    public sealed class Volume : Effect<VolumeParameters>
+    {
+        /// <summary>
+        /// The new volume level... 0 = silent, 1.0 = normal, above 1.0 = amplification. The default value is 1.
+        /// </summary>
+        public float Target
+        {
+            get => Parameters.fTarget;
+            set
+            {
+                Parameters.fTarget = value;
+                OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// The current volume level... -1 = leave existing current level when setting parameters. The default value is 1.
+        /// </summary>
+        public float Current
+        {
+            get => Parameters.fCurrent;
+            set
+            {
+                Parameters.fCurrent = value;
+                OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// The time to take to transition from the current level to the new level, in seconds. The default value is 0.
+        /// </summary>
+        public float Time
+        {
+            get => Parameters.fTime;
+            set
+            {
+                Parameters.fTime = value;
+                OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// The curve to use in the transition... False for linear, true for logarithmic. The default value is false.
+        /// </summary>
+        public bool Curve
+        {
+            get => Convert.ToBoolean(Parameters.lCurve);
+            set
+            {
+                Parameters.lCurve = (uint)(value == true ? 1 : 0);
```

Comment:
redundant comparison of boolean to boolean constant

```suggestion
                Parameters.lCurve = (uint)(value ? 1 : 0);
```

Reply:
So smoogi wants to see if we can use the [Volume channel effect instead of VolumeFx](https://github.com/ppy/osu/pull/27793#discussion_r1553041203). Depending on how that goes, this may or may not be needed. I'll wait until that gets resolved before addressing this.

Reply:
I'll address this in a fixup PR to the PR that implemented this. I will merge that branch into this one when finished.